### ISSUE TITLE
Cannot switch context to Chrome Custom Tab

### ISSUE THREAD

**Issue by roddymunro** [appium/appium#13543]
## The problem

We are unable to switch Context to the Chrome Custom Tab context.

## Environment

* Appium version (or git revision) that exhibits the issue: 1.15.1
* Last Appium version that did not exhibit the issue (if applicable): -
* Desktop OS/version used to run Appium: macOS Catalina
* Node.js version (unless using Appium.app|exe): 
* Npm or Yarn package manager: 
* Mobile platform/version under test: Android 9
* Real device or emulator/simulator: Real
* Appium CLI or Appium.app|exe: Appium.app 

## Details

When our Chrome Custom Tab is displayed, we cannot switch context to it. This means we are unable to automate our login page. We get the following error message:

```
>>> driver.switch_to.context(driver.contexts[1])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/appium/webdriver/switch_to.py", line 30, in context
    self._driver.execute(MobileCommand.SWITCH_TO_CONTEXT, {'name': context_name})
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/appium/webdriver/errorhandler.py", line 29, in check_response
    raise wde
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/appium/webdriver/errorhandler.py", line 24, in check_response
    super(MobileErrorHandler, self).check_response(response)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: An unknown server-side error occurred while processing the command. Original error: unknown error: Failed to get sockets matching: @webview_devtools_remote_.*16060
  (make sure the app has its WebView configured for debugging)
```

## Link to Appium logs

https://gist.github.com/roddymunro/1778d2a0c98d1eddb5b0d52225d64dd8

**Comment by mykola-mokhnach** [comment]
I think Appium has never been advertising CCT support. Are you able to see the tab in Chrome remote debugger by the way?

**Comment by roddymunro** [comment]
@mykola-mokhnach I can see it in Chrome dev tools inspector (`chrome://inspect/#devices`) with no issues. 

Are there any plans to support CCT?

**Comment by mykola-mokhnach** [comment]
There are no plans yet

**Comment by sandeepyadavkec** [comment]
@mykola-mokhnach I am also stuck with same issue, unable to login, is there any workaround if Appium is not planning to support chrome custom tabs.

**Comment by Jiurong-Yang** [comment]
Any additional info on this? And I guess the current work around is making sure the website end have ID's on the elements and those IDs propagate over to the native page source. Which I've seen is happening with the WebViews which I believe did not happen before.

**Comment by michesp** [comment]
Hello, is there any update on this? Been encountering issues trying to interact with CCT on both Android and iOS.

**Comment by mikhailnersesov** [comment]
I get the same "Failed to get sockets matching: @webplayer_devtools_remote_.*22681" by trying to switch context from native_app to web_view.
Appium 1.18.2
I believe this issue duplicates: https://github.com/appium/python-client/issues/255
And the solution is: https://stackoverflow.com/a/55899339
Does anyone know if Appium team plans to do something with this?

**Comment by KazuCocoa** [comment]
I tested the custom tab with https://github.com/GoogleChrome/android-browser-helper/tree/main/demos/custom-tabs-example-app . Then, it worked like the below. (Emulator and developers option enabled to debug chrome browser). So, it seems to work for now. (At least, I tested the repository with Appium 1.21.0 and current master)
(Possibly older version may require `chromeOptions` and `androidPackage` to handle chrome package name)

https://gist.github.com/KazuCocoa/708fc5f297e5d60cf9862f930ddcabd2

```
[5] pry(#<AppiumLibCoreTest::DriverTest>)> @driver = @@core.start_driver
=> #<Appium::Core::Base::Driver:0xc7f7de9e752e902 browser="unknown">
[6] pry(#<AppiumLibCoreTest::DriverTest>)> @driver.available_contexts
=> ["NATIVE_APP", "WEBVIEW_chrome"]
[7] pry(#<AppiumLibCoreTest::DriverTest>)> @driver.context = 'WEBVIEW_chrome'
=> "WEBVIEW_chrome"
[8] pry(#<AppiumLibCoreTest::DriverTest>)> @driver.title
=> "Android Developers"
```

According to https://developer.chrome.com/docs/android/custom-tabs/integration-guide/ , the chrome package name can be different in beta and stable. Then, probably it requires to set `"chromeOptions": { "androidPackage": "com.android.beta" }` for example.

---

What the result of https://github.com/appium/appium-uiautomator2-driver#mobile-getcontexts in your case?
It may print all available chrome tabs on the device via brave://inspect/#devices . It should help to investigate your case.
